util/data/msgreply.c

```diff
@@ -197,9 +199,9 @@ rdata_copy(sldns_buffer* pkt, struct packed_rrset_data* data, uint8_t* to,
 		if(*rr_ttl > MAX_NEG_TTL)
 			*rr_ttl = MAX_NEG_TTL;
 	}
-	if(*rr_ttl < MIN_TTL)
+	if(!SERVE_ORIGINAL_TTL && (*rr_ttl < MIN_TTL))
```

Comment:
Should a similar check be added to adjust_ttl() in val_sigcrypt.c?

Reply:
Yes, possibly, although this is a bit tricky. The upward adjustment for `MIN_TTL` is fixable, but the downward adjustment if the expiration time comes before the TTL expires is maybe not something to change, what do you think?

Reply:
I think it is best to have dnssec expiration remove the cache entry; so it can be refreshed with a hopefully correct value.  Otherwise the server would keep serving a dnssec invalid value.  So no change in val_sigcrypt is needed.